test/routes/add-article.test.ts

```diff
@@ -28,4 +28,18 @@ describe('add article', (): void => {
   it('should throw an error if id is already set', async (): Promise<void> => {
     await expect(makeRequest(createArticle('_:1'))).rejects.toThrow(new createHttpError.Forbidden('Article IDs must not be set (\'_:1\' was given)'));
   });
+
+  it('should throw an error if it is not a schema:Article', async (): Promise<void> => {
+    await expect(makeRequest({
+      '@id': '_:1',
+      '@type': 'http://schema.org/NewsArticle',
```

Comment:
interesting case, because this is a subtype of `http://schema.org/Article`. Perhaps use something clearly different like https://schema.org/Book?

Reply:
Useful to include both cases (as well as one that is both an `Article` _and_ `Book`). RDF inheritance is not something I'm that clear on what we should be doing. Seems like inheritance is not normally understood, but you can choose to understand it. (Declaring it as both an `Article` and `NewsArticle` is the more explicit way).

I'm even less clear on _property_ inheritance...

/cc @stephenwf @tpluscode in case they have any words of wisdom

Reply:
(not sure what should happen on a subtype)

Reply:
Interesting to generate this discussion. It was not intended but is valuable. This scenario was just supposed to add a `@type` that was not expected.

Reply:
For now, this is correctly a failing case (`http://schema.org/Article` needs to be explicit). Might change in the future.

Reply:
My understanding of subtyping is that it's an open world where you can't possibly know all of the subtypes B, C, ... of type A; so to allow subtypes to be passed in you would have to resolve somehow B and understand whether it declares A as its supertype. The client could also indicate `@type` contains both the A and B to make it clear.

Reply:
So checklist is:
- `Book` (refuse as different type)
- both `Article` and `Book` (accept as the first type allows it)
- `NewsArticle` (refuse currently, as it's a subtype)